usage-dash-oauth.py

```diff
@@ -0,0 +1,102 @@
+import dash
+from dash.dependencies import Input, Output, State
+import dash_core_components as dcc
+import dash_html_components as html
+
+import binascii
+import json
+import flask
+import os
+
+from dash_auth.dash_oauth import DashTokenAuth
+
+
+server = flask.Flask(__name__)
+server.secret_key = (
+    'fd8e7047cd8c3bedef10568f124aad5102899537a5ec49e15af1d195caf4'
+)
+USERS = [
+    {
+        'username': 'chris',
+        'password': 'snowstorm',
+        'secret-token': binascii.hexlify(os.urandom(24)),
+        'has-access': True
+    },
+    {
+        'username': 'tom',
+        'password': 'rainfall',
+        'secret-token': binascii.hexlify(os.urandom(24)),
+        'has-access': False
+    }
+]
+
+class DashAuth(DashTokenAuth):
+    def __init__(self, app):
+        DashTokenAuth.__init__(self, app)
+
+    def validate_user(self, username, password):
+        print('validate_user')
+        user = [u for u in USERS if u['username'] == username]
+        if len(user) == 0:
+            return (False, '',)
+        else:
+            if user[0]['password'] == password and user[0]['has-access']:
+                return (True, user[0]['secret-token'],)
```

Comment:
If the user has access, then this function is to return the user's secret unique identifier